pipelines/templates/config-version.yml

```diff
@@ -1,19 +1,56 @@
 parameters:
 - name: platform
   type: string
+- name: python
+  type: string
+  default: default
+  values:
+  - default
+  - noop
+  - venv
 
 steps:
-- task: UsePythonVersion@0
-  inputs:
-    versionSpec: 3.9
-  displayName: (latest) Configure Python version
-  condition: and(succeeded(), not(contains('${{ parameters.platform }}', 'legacy')))
+
+# UsePythonVersion task only works when the specific Python version is already installed.
+
+# The following is for linux.
+# Reference: https://dev.to/akaszynski/create-an-azure-self-hosted-agent-without-going-insane-173g
+# We only need Python 3.7 and 3.9 for now.
+# --system-site-packages is required to make packages installed with --user visible to virtualenv.
+- ${{ if contains(parameters.platform, 'legacy') }}:
+    script: |
+      set -e
+      mkdir -p $(Agent.ToolsDirectory)/Python
+      cd $(Agent.ToolsDirectory)/Python
+      PY37_VER=$(python3.7 -c "import sys; print('.'.join([f'{val}' for val in sys.version_info[:3]]))")
+      mkdir $PY37_VER
+      ln -s $PY37_VER 3.7
+      cd $PY37_VER
+      python3.7 -m venv x64 --system-site-packages
+      touch x64.complete
+    displayName: Create Python 3.7 venv
+  ${{ else }}:
+    script: |
+      set -e
+      mkdir -p $(Agent.ToolsDirectory)/Python
+      cd $(Agent.ToolsDirectory)/Python
+      PY39_VER=$(python3.9 -c "import sys; print('.'.join([f'{val}' for val in sys.version_info[:3]]))")
+      mkdir $PY39_VER
+      ln -s $PY39_VER 3.9
```

Comment:
If I upgrade latest to Python 3.10, do I need to update the vmss image?

Reply:
Yes. We didn't install python 3.10 in the image, because I thought we only need python 3.7 and 3.9.

New python versions are installed from deadsnakes/ppa. So it's just a new line of apt-get.